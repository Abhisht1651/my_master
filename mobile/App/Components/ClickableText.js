// @flow
import React from 'react'
import { Text, TouchableOpacity, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'

type Props = {
  onPress?: PropTypes.func,
  textStyle?: StyleSheet,
  children?: string,
  style?: StyleSheet,
  text?: string
}
const ClickableText = (props: Props) => {
  const { style, onPress, textStyle, children, text } = props
  return (
    <TouchableOpacity style={style} onPress={onPress}>
      <Text style={textStyle}> {children || text} </Text>
    </TouchableOpacity>
  )
}

ClickableText.defaultProps = {
  onPress: () => {},
  textStyle: StyleSheet.create({}),
  children: '',
  style: StyleSheet.create({}),
  text: ''
}

export default ClickableText
