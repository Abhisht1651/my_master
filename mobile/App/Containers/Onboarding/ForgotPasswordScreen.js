/* eslint-disable no-unused-vars */
// @flow

import { NavigationContainerProps } from '@react-navigation/native'

import React, { useState } from 'react'
import { Text, Image, View, TouchableOpacity, Platform } from 'react-native'
import { Images } from '../../Themes'

// Styles
import styles from '../Styles/Onboarding/ForgotPasswordScreenStyles'
import InputText from '../../Components/TextInput/TextInputComponent'
import ClickableText from '../../Components/ClickableText'
import { MediumButton } from '../../Components/Buttons/MediumButton'

type Props = NavigationContainerProps & {}
const ForgotPasswordScreen = (props: Props) => {
  const [userInfo, setUserInfo] = useState('')
  return (
    <View style={styles.mainContainer}>
      {Platform.OS === 'ios' && <View style={styles.statusBar} />}

      <View style={styles.mainContainer}>
        <TouchableOpacity
          onPress={() => props.navigation.goBack()}
          style={styles.backArrowContainer}>
          <Image source={Images.leftArrow} style={styles.backArrow} />
        </TouchableOpacity>

        <View style={styles.section}>
          <Image source={Images.newzeraPurpleLogo} style={styles.logo} />
          <Text style={styles.forgotText}>Forgot Password?</Text>
          <Text style={styles.emailText}>Enter your email associated with your account.</Text>
          <View style={styles.inputText}>
            <InputText placeholder="Email /Phone No." captureText={(text) => setUserInfo(text)} />
          </View>
        </View>
        <View style={styles.buttonWrap}>
          <MediumButton title="Continue" onPress={() => props.navigation.navigate('EmailSent')} />

          <View style={styles.sectionRow}>
            <Text style={styles.signUpTextMain}>Not an user? </Text>
            <ClickableText
              textStyle={styles.signUpTextSignUp}
              onPress={() => props.navigation.navigate('SignUp')}
              text="Sign up"
            />
          </View>
        </View>
      </View>
    </View>
  )
}

export default ForgotPasswordScreen
