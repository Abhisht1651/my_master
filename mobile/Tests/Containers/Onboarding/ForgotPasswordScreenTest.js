import { Image, Text } from 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import ForgotPasswordScreen from '../../../App/Containers/Onboarding/ForgotPasswordScreen'
import { MediumButton } from '../../../App/Components/Buttons/MediumButton'
import InputText from '../../../App/Components/TextInput/TextInputComponent'

import ClickableText from '../../../App/Components/ClickableText'

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn()
  },
  ...props
})

describe('ForgotPasswordScreen', () => {
  describe('rendering', () => {
    let wrapper
    let props
    beforeEach(() => {
      props = createTestProps({})
      wrapper = shallow(<ForgotPasswordScreen {...props} />) // no compile-time error
    })
    it('should match to snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should render 2 <Image />', () => {
      expect(wrapper.find(Image)).toHaveLength(2)
    })

    it('should render 1 <InputText />', () => {
      expect(wrapper.find(InputText)).toHaveLength(1)
    })
    it('1st <InputText /> is email', () => {
      const inputText = wrapper.find(InputText).at(0)
      expect(inputText.prop('placeholder')).toBe('Email /Phone No.')
    })
    it('should render 3 <Text /> tag', () => {
      expect(wrapper.find(Text)).toHaveLength(3)
    })

    it('should render 1 <Rounded Button />', () => {
      expect(wrapper.find(MediumButton)).toHaveLength(1)
    })
    it('should navigate to Modal Screen when clicked on Button', () => {
      const button1 = wrapper.find(MediumButton).at(0)
      button1.simulate('press')
      expect(props.navigation.navigate).toHaveBeenCalledWith('EmailSent')
    })

    it('should render <ClickableText /> tag', () => {
      expect(wrapper.find(ClickableText)).toHaveLength(1)
    })
    it('should navigate to SignUp Screen when clicked on SignUp Text', () => {
      const button1 = wrapper.find(ClickableText).at(0)
      button1.simulate('press')
      expect(props.navigation.navigate).toHaveBeenCalledWith('SignUp')
    })
  })
})
