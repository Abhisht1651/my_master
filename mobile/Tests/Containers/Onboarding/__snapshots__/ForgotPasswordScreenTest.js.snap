// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordScreen rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 58.188802083333336,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "left": 54,
          "position": "absolute",
          "top": 2.60546875,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 31.265625,
            "resizeMode": "contain",
            "width": 36,
          }
        }
      />
    </ForwardRef>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 217.04179999999997,
            "marginTop": 133.4,
            "resizeMode": "contain",
            "width": 434.99999999999994,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "BananaGrotesk-Medium",
            "fontSize": 38,
            "marginTop": "3%",
          }
        }
      >
        Forgot Password?
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "BananaGrotesk-Light",
            "fontSize": 28,
            "marginHorizontal": "20%",
            "marginTop": "2%",
            "textAlign": "center",
          }
        }
      >
        Enter your email associated with your account.
      </Text>
      <View
        style={
          Object {
            "marginTop": 118.11458333333334,
            "width": 550,
          }
        }
      >
        <TextInputComponent
          borderBottomColor="#7B46E4"
          borderBottomStyle={Object {}}
          captureText={[Function]}
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="Email /Phone No."
          placeholderTextColor="#8F8FAF"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 88.5776,
        }
      }
    >
      <MediumButton
        disabled={false}
        onPress={[Function]}
        outline={false}
        style={Object {}}
        title="Continue"
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 266.8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 31,
            }
          }
        >
          Not an user? 
        </Text>
        <ClickableText
          onPress={[Function]}
          style={Object {}}
          text="Sign up"
          textStyle={
            Object {
              "color": "#7B46E4",
              "fontFamily": "BananaGrotesk-Bold",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
